osquery/tables/applications/linux/docker_deb_packages.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <sys/socket.h>
+#include <sys/types.h>
+
+#include <boost/algorithm/string.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/applications/linux/namespace_ops.h>
+#include <osquery/tables/applications/posix/docker_api.h>
+#include <osquery/tables/system/linux/deb.h>
+
+
+#include <rapidjson/filereadstream.h>
+
+namespace osquery {
+namespace tables {
+
+const unsigned int BUF_SIZE = 65536;
+
+void extractDebPackageInfo(const struct pkginfo* pkg,
+                           rapidjson::Writer<rapidjson::StringBuffer>& w) {
+  struct varbuf vb;
+  varbuf_init(&vb, 20);
+  w.StartObject();
+
+  // Iterate over the desired fieldinfos, calling their fwritefunctions
+  // to extract the package's information.
+  const struct fieldinfo* fip = nullptr;
+  for (fip = fieldinfos; fip->name; fip++) {
+    fip->wcall(&vb, pkg, &pkg->installed, fw_printheader, fip);
+
+    std::string line = vb.string();
+    if (!line.empty()) {
+      size_t separator_position = line.find(':');
+
+      std::string key = line.substr(0, separator_position);
+      std::string value = line.substr(separator_position + 1, line.length());
+
+      auto it = kFieldMappings.find(key);
```

Comment:
```cpp
if (const auto it = kFieldMappings.find(key); it != kFieldMappings.end()) { }
```

Reply:
+1. I think something along the lines of this and/or using `split`/`join`s might work better?